packages/kolibri/components/DownloadButton.vue

```diff
@@ -38,6 +39,26 @@
       files: {
         type: Array,
         default: () => [],
+        validator: function (files) {
+          return files.every(file =>
+            validateObject(file, {
+              checksum: { type: String, required: true },
+              extension: { type: String, required: true },
+              preset: { type: String, required: true },
+              lang: {
+                type: Object,
+                required: false,
```

Comment:
For objects, you can nest a further spec for the object under the `spec` key. See here for an example: https://github.com/learningequality/kolibri/blob/develop/packages/kolibri/styles/internal/themeSpec.js#L49